File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/_1_ControlFreakRefactoredToDependencyInjection/InMessages/StartMessage.cs
using System;
using DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Interfaces;
using DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Services;

namespace DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.InMessages
{
  class StartMessage : AcmeMessage
  {
    public void AuthorizeUsing(IAuthorization authorizationRules)
    {
      Console.WriteLine("Authorizing Start with " + authorizationRules);
    }

    public void WriteTo(DataDestination dataDestination)
    {
      dataDestination.Add("Writing Start to " + dataDestination);
    }
  }
}

File: Java/dependency-injection/_03_control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/InMessages/StartMessage.java
package com.github.grzesiek_galezowski.DependencyInjectionAfter.InMessages;

import com.github.grzesiek_galezowski.DependencyInjectionAfter.Interfaces.AcmeMessage;
import com.github.grzesiek_galezowski.DependencyInjectionAfter.Interfaces.DataDestination;
import com.github.grzesiek_galezowski.DependencyInjectionAfter.Services.Authorization;

public class StartMessage implements AcmeMessage {
    public void authorizeUsing(Authorization authorizationRules) {
        System.out.println("Authorizing start with " + authorizationRules);
    }

    public void writeTo(DataDestination dataDestination) {
        dataDestination.add("Writing start to " + dataDestination);
    }
}
